code/modules/ninja/martial_art.dm

```diff
@@ -0,0 +1,182 @@
+ * Creeping Widow MMB override datum
+ * Creeping Widow injector
+ */
+
+
+// Creeping Widow injector - Single use nanomachine thing that teaches people the creeping widow style.
+
+
+/obj/item/weapon/creeping_widow_injector/
+	name = "strange injector"
+	desc = "A strange autoinjector made of a black metal.<br>You can see a green liquid through the glass."
+	icon = 'icons/obj/ninjaobjects.dmi'
+	icon_state = "injector"
+	attack_verb = list("poked", "prodded")
+	var/used = 0
+
+/obj/item/weapon/creeping_widow_injector/attack_self(mob/living/carbon/human/user as mob)
+	if(!used)
+		user.visible_message("<span class='warning'>You stick the [src]'s needle into your arm and press the button.", \
+			  "<span class='warning'>[user] sticks the [src]'s needle \his arm and presses the button.")
+		user << "<span class='info'>The nanomachines in the [src] flow through your bloodstream."
+
+		var/datum/martial_art/ninja_martial_art/N = new/datum/martial_art/ninja_martial_art(null)
+		N.teach(user)
+
+		used = 1
+		icon_state = "injector-used"
+		desc = "A strange autoinjector made of a black metal.<br>It appears to be used up and empty."
+		return 0
+	else
+		user << "<span class='warning'>The [src] has been used already!</span>"
+		return 1
+
+// Ninja martial art datum
+
+/datum/martial_art/ninja_martial_art
+	name = "Creeping Widow Style"
+	var/list/attack_names = list("dragon", "eagle", "mantis", "tiger", "spider", "monkey", "snake", "crane", "xeno") // Fluff attack texts, used later in attack message generation.
+	var/has_choke_hold = 0 	// Are we current choking a bitch?
+	var/has_focus = 1		//Can we user our special moves?
+
+/datum/martial_art/ninja_martial_art/teach(var/mob/living/carbon/human/H,var/make_temporary=0)
+	..()
+	H.middleClickOverride = new /datum/middleClickOverride/ninja_martial_art()
+	H << 	"You have been taugh the ways of the <i>Creeping Widow</i>.<br>\
+			Your stikes on harm intent will deal more damage.<br>Using middle mouse button on a nearby person while on harm intent will send them flying backwards.<br>\
+			Your grabs will instantly be aggressive while you are using this style.<br>Using middle mouse button while on harm intent and behind a person will put them in a silencing choke hold.<br>\
+			Using middle mouse button on a nearby person while on disarm intent will wrench their wrist, causing them to drop what they are holding.</span>"
+
+/datum/martial_art/ninja_martial_art/proc/wrist_wrench(var/mob/living/carbon/human/A, var/mob/living/carbon/human/D)
+	if(!D.stat && !D.weakened)
+		if(has_focus)
+			has_focus = 0
+			A.face_atom(D)
+			D.visible_message("<span class='warning'>[A] grabs [D]'s wrist and wrenches it sideways!</span>", \
+							  "<span class='userdanger'>[A] grabs your wrist and violently wrenches it to the side!</span>")
+			playsound(get_turf(A), 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
+			D.emote("scream")
+			D.drop_item()
+			D.apply_damage(5, BRUTE, pick("l_arm", "r_arm"))
+			D.Stun(1)
```

Comment:
stuns make you drop what's in your active hand, so having drop_item() and stun here is redundant.